### PR TITLE
Update building-from-source.md

### PR DESCRIPTION
**Pull request by jpmoniz**
Fix dependency issues

### CODE REVIEW

File: docs/building-from-source.md [hunk 1]
```diff
@@ -93,18 +94,21 @@ result in a root-owned file being left in the machinekit directory.
     sudo apt-get install git dpkg-dev
```
Comment (zultron) [inline docs/building-from-source.md]: Can you add `devscripts`, the package that provides `mk-build-deps`, here?

File: docs/building-from-source.md [hunk 2]
```diff
@@ -93,18 +94,21 @@ result in a root-owned file being left in the machinekit directory.
...
-    debian/configure -px
-    sudo scripts/apt-installbuilddeps
+    debian/configure -x
+    sudo mk-build-deps -ir
     cd src
     ./autogen.sh
     ./configure
     make
     sudo make setuid
+    sudo adduser <username> <kernel Flavor>
```
Comment (zultron) [inline docs/building-from-source.md]: I think this only applies to `xenomai` threads.

**Comment by RunningLight** [comment]
* using `apt-get build-dep` (which is already installed by `dpkg-dev`) works if I add `deb-src http://deb.dovetail-automata.com wheezy main` to the repo list, say as a second line in `/etc/apt/sources.list.d/machinekit.list` whereas `mk-build-deps` works without such help if we add more packages by installing `devscripts`. I suppose the latter approach is better because it doesn't depend on the packages in the dovetail-automata repo being in sync with the sources in the git repo, but is that really a possibility? [bottom line: I can go either way on this one]

* Thanks for picking up that `-r` option on `mk-build-deps`. I wasn't aware of it when I wrote the original version of these docs. Without it, one is left with the deb package owned by root sitting in the machinekit directory. Hence the note which was added when I changed the instructions to use `scripts/apt-installbuilddeps`. Whatever we do now, that note is no longer relevant and I'll remove it in my rewrite (mentioned below)

* I understand adding the user to the xenomai group is necessary but that should be in the instructions for installing xenomai, right? Why here?

* `configure -x` specifically does not build with `posix` threads. I was already rewriting this doc to explain more of the options but I lean toward retaining the `-p` option in the instructions for newbies. A xenomai-only version will build on a non-realtime host, but funny things happen if one tries to run the resulting simulators on it. For fun, I just tried it in a VM---the VM crashed with Oracle VirtualBox reporting "Aborted".

**Comment by jpmoniz** [comment]
@RunningLight 

Weird My VM is running the Sim configs just fine.  I edited the machinekit demo gcode to loop 10 times and ran with no problems.

As far as where to add to the xenomai user group I'm indifferent either way.

**Comment by RunningLight** [comment]
Weird indeed. It works for me on a actual host but not on a virtual host.

Would you entertain two changes?

* restore `debian/configure -px` for now so there's less surprise to readers who sorta remember what it said before and we skirt VM issues. As the teaser line at the bottom of the existing page says, I intended to add more explanation and I will.

* modify the proposed change

  ```
+if configure was ran with the option --with-xenomai else skip
+
+    sudo adduser <username> xenomai
```

  As written, it mixes up the use of options in the first `debian/configure` (with its `-x` option) and the second `./configure` (with its optional `--with-xenomai`). Also, "else skip" may be a little cryptic to some; this isn't code here. Also, there may well be hosts set up with multiple user accounts. Might I suggest we say something like

  > Users who wish to invoke machinekit (built with xenomai threads enabled) on a xenomai realtime kernel must ensure they are members of the xenomai group. If that wasn't already done when installing the kernel, then add each such user now

  > `sudo adduser <username> xenomai`

I'm willing to merge your PR either way, but this way I don't propose these changes as soon as it's merged :smile:

**Comment by jpmoniz** [comment]
Yes I agree with you.  I wont be in a position to test until tonight

**Comment by zultron** [comment]
On 05/07/2015 07:26 AM, Kent Reed wrote:
>   *
>
>     I suppose the [`mk-build-deps`] approach is better [than
 >     `apt-get build-dep`] because it
>     doesn't depend on the packages in the dovetail-automata repo being
>     in sync with the sources in the git repo, but is that really a
>     possibility?

It's a possibility, but in reality, the `Build-Depends:` list doesn't 
change often.  If using the Dovetail Automata repo with automated 
updates and building `master`, there's a < 1 hour window in which 
build-deps would be out of sync.

The `mk-build-deps` approach would work seamlessly for that < 1 hour, 
for MK distributions not synched with `master`, and for building dev 
branches with modified build-deps.  But installing a dep or two by hand 
shouldn't be a big deal for someone wishing to build from source, so 
this isn't a major problem.

>   *
>
>     I understand adding the user to the xenomai group is necessary but
>     that should be in the instructions for installing xenomai, right?
>     Why here?

I don't see separate instructions for installing Xenomai on that page. 
Pointer?

IMO, that page should contain only brief as possible but complete 
instructions for building MK from source with Xenomai threads for the 
first time from a clean Debian install.

>   *
>
>     |configure -x| specifically does not build with |posix| threads. I
>     was already rewriting this doc to explain more of the options but I
>     lean toward retaining the |-p| option in the instructions for
>     newbies.

Either way is fine with me.  A reason *not* to build posix threads is 
there can be no mistake when the user expects hard RT constraints but 
for whatever reason the runtime scripts back down to posix.

> A xenomai-only version will build on a non-realtime host,
>     but funny things happen if one tries to run the resulting simulators
>     on it. For fun, I just tried it in a VM---the VM crashed with Oracle
>     VirtualBox reporting "Aborted".

Interesting, this sounds like a bug.  The code usually does a reasonable 
job choosing the 'best' threads modules given the capabilities of the 
runtime environment, but perhaps we overlooked the case where *no* 
appropriate threads modules match the runtime environment?

**Comment by RunningLight** [comment]
@zultron 

>>  * I understand adding the user to the xenomai group is necessary but that should be in the instructions for installing xenomai, right? Why here?

> I don't see separate instructions for installing Xenomai on that page. Pointer?

Good point. The instructions are on the companion page http://www.machinekit.io/docs/packages-debian/ However, Looking at it now, I see it covers installation of the xenomai-enabled kernel while not mentioning adding users to the xenomai group!?! Is this last step handled automagically when installing Debian packages? Obviously, both pages need some spiffing up (and I haven't responded to my RPi2 issue machinekit/machinekit.github.io#42 yet either).

>>  * `configure -x` specifically does not build with `posix` threads. I was already rewriting this doc to explain more of the options but I lean toward retaining the `-p` option in the instructions for newbies.

> Either way is fine with me. A reason *not* to build posix threads is there can be no mistake when the user expects hard RT constraints but for whatever reason the runtime scripts back down to posix.

Interesting point of view which I hadn't considered. Not building posix threads is fine with me if we can work out my VM issue below.

> > A xenomai-only version will build on a non-realtime host, but funny things happen if one tries to run the resulting simulators on it. For fun, I just tried it in a VM---the VM crashed with Oracle VirtualBox reporting "Aborted".

> Interesting, this sounds like a bug. The code usually does a reasonable job choosing the 'best' threads modules given the capabilities of the runtime environment, but perhaps we overlooked the case where *no* appropriate threads modules match the runtime environment?

I'm still looking for the dog which didn't bark in the night. Now that I've convinced myself that a real host, in this instance a Cubox-i4Pro running vanilla Wheezy, keeps on trucking when I invoke a machinekit built with only xenomai threads enabled, I assume my VM crashing has something to do with the way the Wheezy kernel interacts with the virtual hardware presented by Oracle VirtualBox.

**Comment by RunningLight** [comment]
@jpmoniz  Given the most recent list discussion, It appears we should amend the instructions further to read

```
sudo apt-get install -no-install-recommends devscripts
sudo apt-get install equivs
```

Looking at the multiple invocations of `sudo apt-get install` I have the feeling we could consolidate but I haven't tested to make sure this doesn't introduce a new problem.

**Comment by kinsamanka** [comment]
On 8 May 2015 6:09 am, "Kent Reed" <notifications@github.com> wrote:
>
> @jpmoniz Given the most recent list discussion, It appears we should
amend the instructions further to read
>
> sudo apt-get install -no-install-recommends devscripts
> sudo apt-get install equivs
>

It can be shortened to:

sudo apt-get install -no-install-recommends devscripts equivs

> Looking at the multiple invocations of sudo apt-get install I have the
feeling we could consolidate but I haven't tested to make sure this doesn't
introduce a new problem.

**Comment by jpmoniz** [comment]
@RunningLight 

Please review.  Apologies for my poor git foo

**Comment by RunningLight** [comment]
Later today...and no apologies needed. I was a complete doofus with git at first and now a dilettante at best.

File: docs/building-from-source.md~ [hunk 3]
```diff
@@ -0,0 +1,152 @@
+---
```
Comment (zultron) [inline docs/building-from-source.md~]: Looks like there's an emacs backup file committed here.

File: docs/building-from-source.md [hunk 4]
```diff
@@ -91,20 +92,27 @@ Note: The following steps have been modified and no longer
 result in a root-owned file being left in the machinekit directory.
 
     sudo apt-get install git dpkg-dev
+	sudo apt-get install --no-install-recommends devscripts equivs
```
Comment (zultron) [inline docs/building-from-source.md]: nit pick: indentation

**Comment by RunningLight** [comment]
Didn't want to hold this up any more. merging now and I'll take care of editorial nits (generally, not just this PR) tomorrow night. WWII warbirds gathered here D.C. have my attention today.

**Comment by mhaberler** [comment]
read that article. I'll take the Lightning for a spin ;)

**Pull request merged**